### PR TITLE
Define error code

### PR DESCRIPTION
**Pull request by ZenoTan**
### What problem does this PR solve?
Assign error code to each type of log errors in pd. Closes https://github.com/pingcap/pd/issues/2449.

### What is changed and how it works?
PD error code is from 9800 to 10000, and we reserve double code space for future use.
Log errors are classified by types, such as storage, io, interal... Each error code has its name by its class plus some details.
For example, StorageEtcdLoadErr means it is related to storage, uses etcd, happens when users try to load something.

### Check List

Tests

- No code

Code changes

- New pkg for error code definitions.

Side effects

Related changes

### CODE REVIEW

File: pkg/errno/errno.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package errno
+
+var (
```
Comment (HunDunDM) [inline pkg/errno/errno.go:3]: const?

File: pkg/errno/errno.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package errno
```
Comment (disksing) [inline pkg/errno/errno.go:1]: add license header

**Comment by nolouch** [comment]
@ZenoTan  Please make ci happy.

File: pkg/errors/errors.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
+package errors
```
Comment (nolouch) [inline pkg/errors/errors.go:1]: Add copyright information.

File: pkg/errors/errors_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
+package errors
```
Comment (nolouch) [inline pkg/errors/errors_test.go:1]: Add copyright information.

File: pkg/errors/errors.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	ClassFormat = reg.RegisterErrorClass(5, "format")
+	// ClassOther is other error class
+	ClassOther = reg.RegisterErrorClass(6, "other")
+)
+
+var (
+	// ErrIOJoinConfig is io join config error
+	ErrIOJoinConfig = ClassIO.DefineError().TextualCode("ErrIOJoinConfig").MessageTemplate("config io error during join").Done()
+	// ErrHTTPContentConfig is http content config error
+	ErrHTTPContentConfig = ClassIO.DefineError().TextualCode("ErrIOHTTPContent").MessageTemplate("http content io error").Done()
```
Comment (rleungx) [inline pkg/errors/errors.go:25]: It's better to make this error more clear.

File: pkg/errors/errors.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	// InternalRuleDuplicate is duplicate rule
+	InternalRuleDuplicate = ClassInternal.DefineError().TextualCode("InternalRuleDuplicate").MessageTemplate("duplicate rule found").Done()
+	// InternalRuleMismatch is rule mismatch
+	InternalRuleMismatch = ClassInternal.DefineError().TextualCode("InternalRuleMismatch").MessageTemplate("rule key mismatch").Done()
+	// InternalOperatorNotFound is operator not found
+	InternalOperatorNotFound = ClassInternal.DefineError().TextualCode("InternalOperatorNotFound").MessageTemplate("operator not found").Done()
+	// InternalOperatorOrphan is operator orphan
+	InternalOperatorOrphan = ClassInternal.DefineError().TextualCode("InternalOperatorOrphan").MessageTemplate("operator is orphan").Done()
+	// ErrInternalOperatorMerge is operator merge error
+	ErrInternalOperatorMerge = ClassInternal.DefineError().TextualCode("ErrInternalOperatorMerge").MessageTemplate("operator cannot merge").Done()
```
Comment (rleungx) [inline pkg/errors/errors.go:69]: What does "operator cannot merge" mean?

Comment (ZenoTan) [reply pkg/errors/errors.go:69]: I will change it to "merge operator should be pair".

File: go.mod [hunk 7]
```diff
@@ -45,8 +46,10 @@ require (
...
 	go.etcd.io/etcd v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738
 	go.uber.org/goleak v0.10.0
-	go.uber.org/zap v1.13.0
+	go.uber.org/zap v1.15.0
 	golang.org/x/tools v0.0.0-20200527183253-8e7acdbce89d
 	google.golang.org/grpc v1.25.1
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0
 )
+
+replace github.com/pingcap/errors => github.com/YuJuncen/errors v0.8.1-0.20200727030200-a512ae64de43
```
Comment (nolouch) [inline go.mod:55]: Does the branch already merge to master?

Comment (rleungx) [reply go.mod:55]: No, after that PR is merged, we can update the latest master

**Pull request merged**